src/charts/donut/donut-pct-chart-directive.js

```diff
@@ -0,0 +1,354 @@
+       'tooltipFn': function (d) {
+         return '<span class="donut-tooltip-pf"style="white-space: nowrap;">' +
+                   d[0].value + ' ' + d[0].name +
+                '</span>';
+       },
+       'centerLabelFn': function (scope) {
+         return '<tspan dy="0" x="0" class="donut-title-big-pf">' +
+                  scope.data.available + '</tspan>' +
+                '<tspan dy="20" x="0" class="donut-title-small-pf">' +
+                  'Free' +
+                '</tspan>';
+       }
+     };
+
+     $scope.custData = {
+         'used': '670',
+         'total': '1000'
+     };
+
+   };
+ </file>
+ </example>
+ */
+angular.module('patternfly.charts').directive('pfDonutPctChart', ['ChartsMixin', '$timeout',
+  function(chartsMixin, $timeout) {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        id: '@',
+        config: '=',
+        data: '=',
+        centerLabel: '@'
+      },
+      replace: true,
+      templateUrl: 'charts/donut/donut-pct-chart.html',
+      controller: ['$scope',
+        function($scope) {
+
+          $scope.donutChartId = 'donutChart';
+          if ($scope.id) {
+            $scope.donutChartId = $scope.id + $scope.donutChartId;
+          }
+
+          $scope.updateAvailable = function(){
+            $scope.data.available = $scope.data.total - $scope.data.used;
+          };
+
+          if ($scope.data.available === undefined) {
+            $scope.updateAvailable();
+          }
+
+          $scope.getStatusColor = function(used, thresholds) {
+            if (!thresholds){
+              return '#0088CE';
+            } else {
+              if (used >= thresholds.error) {
+                return '#CC0000';
+              } else if (used >= thresholds.warning) {
+                return '#EC7A08';
+              } else {
+                return '#0088CE';
+              }
+            }
+          };
+
+          $scope.statusDonutColor = function(scope) {
+            var color = {
+              pattern: []
+            };
+            var percentUsed = scope.data.used / scope.data.total * 100.0;
+            color.pattern[0] = $scope.getStatusColor(percentUsed, scope.config.thresholds);
+            color.pattern[1] = '#D1D1D1';
+            return color;
+          };
+
+          var donutTooltip = function(scope) {
+            return {
+              contents: function (d) {
+                if (scope.config.tooltipFn) {
+                  var tootipHtml = '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+                                        scope.config.tooltipFn(d) +
+                                   '</span>';
+                  return tootipHtml;
+                } else {
+                  return '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+                            Math.round(d[0].ratio * 100) + '%' + ' ' + $scope.config.units + ' ' + d[0].name +
+                         '</span>';
+                }
```

Comment:
We've been wanting to speak to you about the .jshintrc for the ridged indentation as well ;-)  
It makes commenting out code for testing really difficult.  I guess I just never worked on a project where there was such close attention to formatting.  I can understand best practices for naming conventions, var declarations, etc., maybe even indentation, but spaces in method signatures!?   IMO this is not an efficient use of our time and it really doesn't make the code any more readable.